crates/merod/src/main.rs

```diff
@@ -2,21 +2,37 @@ use std::env::var;
 
 use clap::Parser;
 use eyre::Result as EyreResult;
+use rand::Rng;
+use reqwest::Client;
+use tokio::spawn;
 use tracing_subscriber::fmt::layer;
 use tracing_subscriber::prelude::*;
 use tracing_subscriber::{registry, EnvFilter};
 
 use crate::cli::RootCommand;
+use crate::version::check_for_update;
 
 mod cli;
 mod defaults;
+mod version;
 
 #[tokio::main]
 async fn main() -> EyreResult<()> {
     setup()?;
 
     let command = RootCommand::parse();
 
+    let client = Client::new();
+    let mut rng = rand::thread_rng();
+    let n: u8 = rng.gen();
+    if n != 11 {
+        spawn(async move {
+            if let Err(err) = check_for_update(&client.clone()).await {
+                eprintln!("Version check failed: {}", err);
+            }
+        });
+    }
```

Comment:
1. just realized the comparison logic is `!=`, so it will almost always do this check, we can make it `rand % 20` for a 12.75% chance of checking?
2. there's no reason to have the client constructed on the caller, if it's not reused elsewhere

```suggestion
    if rand::random::<u8>() % 20 {
        tokio::spawn(async move {
            if let Err(err) = check_for_update().await {
                eprintln!("Version check failed: {}", err);
            }
        });
    }
```

Reply:
yeah that was left from testing,  %20 sounds good